faceted_search/src/index.js

```diff
@@ -1,65 +1,168 @@
 import ReactDOM from 'react-dom';
 import reportWebVitals from './reportWebVitals';
 
 import {
-	SolrFacetedSearch,
-	SolrClient
+    SolrFacetedSearch,
+    SolrClient
 } from "solr-faceted-search-react";
 
 import solrReducer from "./solr-reducer";
-import { createStore } from "redux"
+import { createStore } from "redux";
+
+// react router to define url
+import { 
+    BrowserRouter,
+    Routes,
+    Route,
+    useSearchParams,
+} from 'react-router-dom';
+
+// encode and decode parameter
+import { encode, decode } from "plantuml-encoder"
 
 const config = require("./config.json")
 // Create a store for the reducer.
 const store = createStore(solrReducer);
+
 // The search fields and filterable facets you want
 const fields = [
-	{label: "All text fields", field: "*", type: "text"},
-	{label: "Source", field: "source", type: "list-facet", facetSort:"index"},
-	{label: "Context", field: "hasContextCategory", type: "list-facet", facetSort:"count"},
-	{label: "Material", field: "hasMaterialCategory", type: "list-facet", facetSort:"count"},
-	{label: "Specimen", field: "hasSpecimenCategory", type: "list-facet", facetSort:"count"},
-	{label: "Registrant", field: "registrant", type: "list-facet", facetSort:"count"},
+    {label: "All text fields", field: "searchText", type: "text"},
+    {label: "Identifier", field: "id", type: "text"},
+    {label: "Source", field: "source", type: "list-facet", facetSort:"index"},
+    {label: "Context", field: "hasContextCategory", type: "list-facet", facetSort:"count"},
+    {label: "Material", field: "hasMaterialCategory", type: "list-facet", facetSort:"count"},
+    {label: "Specimen", field: "hasSpecimenCategory", type: "list-facet", facetSort:"count"},
+    {label: "Registrant", field: "registrant", type: "list-facet", facetSort:"count"},
 ];
 
 // The sortable fields you want
 const sortFields = [
-	{label: "Name", field: "koppelnaam_s"},
-	{label: "Date of birth", field: "birthDate_i"},
-	{label: "Date of death", field: "deathDate_i"}
+    {label: "Identifier", field: "id"},
+    {label: "Registrant", field: "registrant"},
 ];
 
 
 // Construct the solr client api class
-const solrClient = 	new SolrClient({
-	url: config.solr_url,
-	searchFields: fields,
-	sortFields: sortFields,
+const solrClient = new SolrClient({
+    url: config.solr_url,
+    searchFields: fields,
+    sortFields: sortFields,
+    rows: 20,
+    pageStrategy: "paginate",
+    start: 0,
 
-	// Delegate change callback to redux dispatcher
-	onChange: (state) => store.dispatch({type: "SET_SOLR_STATE", state: state})
+    // Delegate change callback to redux dispatcher
+    onChange: (state) => {store.dispatch({type: "SET_SOLR_STATE", state: state})}
 });
 
+
+
+function APP() {
+	// https://reactrouterdotcom.fly.dev/docs/en/v6/api#usesearchparams
+	// Used for modifying the query string
+    let [searchParams, setSearchParams] = useSearchParams();
+
+	// https://reactjs.org/docs/hooks-state.html
```

Comment:
We can get rid of these three lines since we're no longer using this API.

Reply:
But we still use this API to modify the URL parameter every time the redux store changes.

Reply:
OK, maybe move the comment with the link to the documentation down to where we use the API and get rid of the two lines we aren't using?

Reply:
OK, thank you for your suggestions!